immutablecollections/_immutableset.py

```diff
@@ -89,9 +92,14 @@ def immutableset(
     iteration_order = []
     containment_set: MutableSet[T] = set()
     for value in iterable:
-        if value not in containment_set:
-            containment_set.add(value)
-            iteration_order.append(value)
+        if value in containment_set:
```

Comment:
Maybe instead of `continue`, keep the `not in` condition and have `else if forbid_duplicate_elements`?  It's just a matter of style, but I like to put the "happy path" up front and to avoid `break/continue` when a conditional will do.

Reply:
Oh that's better. More intuitive.